src/cargo/core/workspace.rs

```diff
@@ -1657,70 +1658,39 @@ pub struct InheritableFields {
     keywords: Option<Vec<String>>,
     categories: Option<Vec<String>>,
     license: Option<String>,
+    #[serde(rename = "license-file")]
     license_file: Option<String>,
     repository: Option<String>,
     publish: Option<VecStringOrBool>,
     edition: Option<String>,
     badges: Option<BTreeMap<String, BTreeMap<String, String>>>,
+    #[serde(rename = "rust-version")]
     rust_version: Option<String>,
+    // We use skip here since it will never be present when deserializing
+    // and we don't want it present when serializing
+    #[serde(skip)]
     ws_root: PathBuf,
 }
 
 impl InheritableFields {
-    pub fn new(
-        dependencies: Option<BTreeMap<String, TomlDependency>>,
-        version: Option<semver::Version>,
-        authors: Option<Vec<String>>,
-        description: Option<String>,
-        homepage: Option<String>,
-        documentation: Option<String>,
-        readme: Option<StringOrBool>,
-        keywords: Option<Vec<String>>,
-        categories: Option<Vec<String>>,
-        license: Option<String>,
-        license_file: Option<String>,
-        repository: Option<String>,
-        publish: Option<VecStringOrBool>,
-        edition: Option<String>,
-        badges: Option<BTreeMap<String, BTreeMap<String, String>>>,
-        rust_version: Option<String>,
-        ws_root: PathBuf,
-    ) -> InheritableFields {
-        Self {
-            dependencies,
-            version,
-            authors,
-            description,
-            homepage,
-            documentation,
-            readme,
-            keywords,
-            categories,
-            license,
-            license_file,
-            repository,
-            publish,
-            edition,
-            badges,
-            rust_version,
-            ws_root,
-        }
+    pub fn update_ws_path(&mut self, ws_root: PathBuf) {
+        self.ws_root = ws_root;
     }
 
     pub fn dependencies(&self) -> CargoResult<BTreeMap<String, TomlDependency>> {
         self.dependencies.clone().map_or(
-            Err(anyhow!("`workspace.dependencies` was not defined")),
+            Err(anyhow!("`workspace.package.dependencies` was not defined")),
```

Comment:
My expectation is dependencies stay as `workspace.dependencies`.  Think of this as mirroring the tables that they come from in `Cargo.toml`.

Reply:
Thats works for me. I will have to add an update function similar to
```rust
pub fn update_deps(&mut self, deps: Option<BTreeMap<String, TomlDependency>>) {
    self.deps = deps;
}
```
and make `dependencies` `#[serde(skip)]` then in `InheritableFields`